src/mysky/dac.ts

```diff
@@ -0,0 +1,16 @@
+import { Permission } from "skynet-interface-utils";
+import { SkynetClient } from "../client";
+
+import { Connector, CustomConnectorOptions } from "./connector";
+
+export abstract class DacLibrary {
+  protected connector?: Connector;
+
+  public constructor(protected dacDomain: string) {}
+
+  public async init(client: SkynetClient, customOptions: CustomConnectorOptions) {
+    this.connector = await Connector.init(client, this.dacDomain, customOptions);
+  }
+
+  abstract getPermissions(): Promise<Permission[]>;
```

Comment:
So here it's defined on the `DacLibrary`, this has my vote as well, but we should explicitly sign off on this then once and for all. I'll bring it up in Discord again as well.

Reply:
I'm assuming this is async because you want it to call the DAC internally? Couldn't we then instead of making it `abstract` just implement it? Don't we want this to be non async, so we don't have to wait for the handshake of every DAC.

Reply:
As discussed this will be on the DacLibrary and not `async`.